tests/specs/lemmas.k

```diff
@@ -5,6 +5,34 @@ module LEMMAS
     imports LEMMAS-JAVA
     imports LEMMAS-HASKELL
 
+    rule WS ++ .ByteArray => WS  [simplification]
+    rule .ByteArray ++ WS => WS  [simplification]
+
+    //Byte array store operations sorting. Slices with lower index first. Helps applying rule below.
+    rule MEM [ N := BUF:ByteArray ] [ M := BUF' ] => MEM [ M := BUF' ] [ N := BUF ]
+      requires N >Int M andBool N -Int M >=Int #sizeByteArray(BUF')                                               [simplification]
+
+    rule M [ N := BUF ] [ N := BUF' ] => M [ N := BUF' ] requires #sizeByteArray(BUF) ==Int #sizeByteArray(BUF')  [simplification]
+    //when BUF1 and BUF2 partially overlap and cannot be sorted by sorting rule.
+    rule MEM [ N := BUF1 ] [ M := BUF2:ByteArray ] [ N := BUF3 ] => MEM [ M := BUF2 ] [ N := BUF3 ]
+      requires #sizeByteArray(BUF1) ==Int #sizeByteArray(BUF3)                                                    [simplification]
+
+    rule BUF [ L .. W ] => .ByteArray requires W <=Int 0                    [simplification]
+    rule BUF [ 0 .. W ] => BUF        requires W ==Int #sizeByteArray(BUF)  [simplification]
+    rule BUF [ L .. W ] => .ByteArray requires L  >Int #sizeByteArray(BUF)  [simplification]
```

Comment:
We can't add this third rule to both modules because the semantics are different for Java and Haskell implementation. Java implementation will return the empty `.ByteArray` like this says, and Haskell implementation will return the zero-padded bytearray of size `W`.

Reply:
done. Separate lemmas for Java and Haskell